apps/nextjs/src/components/post.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+interface PostProps {
+  data: {
+    id: number;
+    title: string;
+    contents: string;
+    createdAt: Date;
+  };
+  comments: {
+    id: number;
+    postId: number;
+    text: string;
+    createdAt: Date;
+  }[];
+}
```

Comment:
You can infer the type for the comments and posts, so that if they change you don't need to make a FE change: https://trpc.io/docs/client/vanilla/infer-types